interfaces/auth_proxy/v0/README.md

```diff
@@ -0,0 +1,58 @@
+# `auth_proxy`
+
+## Usage
+
+This relation interface describes the expected behavior of charms claiming to be able to interact with Policy Decision Point (Oathkeeper), which forms part of Identity and Access Proxy (IAP).
+
+## Direction
+
+The interface will consist of a provider and a requirer.
+
+The requirer is expected to supply configuration required to set up an Identity and Access Proxy: protected urls, allowed endpoints and headers.
+
+The provider will read the information from the application databag, transform them into access rules and forward relevant configuration to Traefik or other API Gateway. It will not write to the application databag.
+
+```mermaid
+flowchart
+    Requirer -- protected_urls, allowed_endpoints, headers --> Provider
+    Provider ----> Requirer
+```
+
+## Behavior
+
+Both the requirer and the provider need to adhere to a certain set of criteria to be considered compatible with the interface:
+
+### Provider
+
+- Is expected to serve Access Control Decision API
+- Is expected to provide a Policy Decision Point in IAP architecture
+- Is expected to transform incoming `auth_proxy` configuration into access rules, enforcing user authentication for requests hitting the protected urls, with the exception of allowed endpoints.
+
+### Requirer
+
+- Is expected to provide configuration required to set up access rules: protected urls, allowed endpoints and headers.
+
+## Relation Data
+
+### Provider
+
+[\[JSON Schema\]](./schemas/provider.json)
+
+n/a
+
+### Requirer
+
+[\[JSON Schema\]](./schemas/requirer.json)
+
+
+#### Example
+
+```json
+{
+  "application_data": {
+    "protected_urls": ["https://10.64.140.43/unit-0", "https://10.64.140.43/unit-1"],
```

Comment:
The example is a bit strange, do we expect the leader to retrieve all of its unit IP's and provide those in its application databag? This seems better suited for placement in unit databag. A bit like what is done with ingress where there is:
- ingress_per_app
- ingress_per_unit

Reply:
We expect that the app requesting to be protected will be related via `ingress-per-app/leader/unit` and will provide its ingress url(s). In case it was related with `ingress-per-unit`, it should retrieve all of its units' ingress urls.

Reply:
Is this explained anywhere?

Reply:
In the [IAP interfaces design](https://docs.google.com/document/d/13a3OEQ_Q4Fwvv9F57wgBYJm4SUFQz4W5M32UVqye4Rg/edit?usp=sharing)

Reply:
Sure but people will read this document here, not the spec you referred. I suggest adding any knowledge related to using this interface here.

Reply:
@natalian98 any update on this one? It's been a week since my last comment here.

Reply:
I pushed some updates, apologies for the delay